### PR TITLE
Updates for chef-ingredient 0.18+

### PR DESCRIPTION
**Pull request by schisamo**
Changes include:

* Moves to using the latest `chef_ingredient` resource which uses mixlib-install under the covers and retrieves artifacts from packages.chef.io. This removes all PackageCloud dependencies.
* Deprecates the use of `node['delivery_build']['repo_name']` and prefers the more aptly named `node['delivery_build']['release-channel']`.
* Deletes a lot of legacy code that is no longer needed!

All code was validated with a `kitchen verify` on the following platforms:

* Ubuntu 14.04
* CentOS 7
* Windows 2012R2

/cc @chef-cookbooks/engineering-services

### CODE REVIEW

File: spec/unit/recipes/chefdk_spec.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@
         end
 
         it 'upgrades chefdk' do
-          expect(chef_run).to upgrade_chef_ingredient('chefdk')
+          expect(chef_run).to install_chef_ingredient('chefdk')
```
Comment (oferrigni) [inline spec/unit/recipes/chefdk_spec.rb]: Does this mean the behavior is changed on reconverge to not upgrade?

**Comment by oferrigni** [comment]
Are there any updates we should make to delivery-cluster as a result of this change?

**Comment by yzl** [comment]
👍 pending @oferrigni's question.

File: spec/unit/recipes/chefdk_spec.rb (hunk 1 above)
Comment (schisamo) [inline spec/unit/recipes/chefdk_spec.rb]: So internally `chef_ingredient` now fetches artifacts with the `remote_file` resource and then installs the artifact with the `package` resource. In this configuration the real indicator of whether an upgrade will occur is tied directly to whether the desired version has changed. You can provide a version via the `version` attribute (say `0.12.0`). You can also provide a version of `latest` which allows `chef_ingredient` to resolve that logical version to a real version. If the version that `latest` resolves to changes between CCRs then an upgrade will be done no matter what action you provide.

**Comment by schisamo** [comment]
@oferrigni delivery-cluster should be fine as similar changes were already applied to this product (the chef-ingredient upgrade):
https://delivery.chef.co/e/chef/#/organizations/delivery/projects/delivery-cluster/changes/29a68011-0695-4d2d-b3c5-09bd0d624968/status/acceptance

**Pull request merged**